src/net/http.rs

```diff
@@ -217,16 +93,40 @@ pub(crate) async fn http_handler(
         return handle_preflight();
     }
 
-    let (req, payload) = match parse_payload(req).await {
+    let inbound_route = match req.method() {
+        // should take the second element as path string starts with a delimiter
+        &Method::GET => req.uri().path().split('/').nth(1).unwrap_or("").to_string(),
```

Comment:
Could you explain why?

Reply:
to extract inbound_route in GET method case

`https://komodoproxy.com`/nft-feature`/api/v2.2/0x1f9090aaE28b8a3dCeaDf281B0F12828e676c326/nft/transfers?chain=eth&format=decimal&order=DESC`

targets `https://moralis.io/api/v2.2/0x1f9090aaE28b8a3dCeaDf281B0F12828e676c326/nft/transfers?chain=eth&format=decimal&order=DESC`

In POST request it is easy to extract inbound_route, as its the whole path.
For GET we decided to include inbound_route in Url, not in header or body. 
Which means that we will need to get inbound_route from Url and change Url on the step when we send req to target.

Reply:
The problem is this will not work with the configuration other than `/something` inbound. 

Let's say we have this configuration:

```
inbound_route: String::from_str("/").unwrap(),
outbound_route: "http://localhost:8000".to_string(),
```

or:

```
inbound_route: String::from_str("/something/else").unwrap(),
outbound_route: "http://localhost:8000".to_string(),
```

in this case, `unwrap_or` will be triggered and `inbound_route` become an empty string.

What about doing this: Iterate over the inbound routes from the configuration and check if the request path starts with any of them. Once you find the matching inbound route, extract that part from the request path and add it to the outbound route.

**There is an edge case** in this logic that needs to be handled; if you have inbound routes like `/nft/special` and `/nft` in the configuration, both will return `true` for `starts_with` on a request path like `demo.com/nft/special/1/2/3`. In such cases, if multiple inbound routes return `true` for `starts_with`, we should always pick the longest one.

Reply:
> The problem is this will not work with the configuration other than `/something` inbound.
> 
> Let's say we have this configuration:
> 
> ```
> inbound_route: String::from_str("/").unwrap(),
> outbound_route: "http://localhost:8000".to_string(),
> ```
> 
> or:
> 
> ```
> inbound_route: String::from_str("/something/else").unwrap(),
> outbound_route: "http://localhost:8000".to_string(),
> ```
> 
> in this case, `unwrap_or` will be triggered and `inbound_route` become an empty string.
> 
> What about doing this: Iterate over the inbound routes from the configuration and check if the request path starts with any of them. Once you find the matching inbound route, extract that part from the request path and add it to the outbound route.
> 
> **There is an edge case** in this logic that needs to be handled; if you have inbound routes like `/nft/special` and `/nft` in the configuration, both will return `true` for `starts_with` on a request path like `demo.com/nft/special/1/2/3`. In such cases, if multiple inbound routes return `true` for `starts_with`, we should always pick the longest one.

The easiest way was just include inbound string in Header, then you can have any inbound_route what you want

Reply:
I still can do some changes in kdf side and move proxy inbound route to header

But if you prefer to have /path1/path2/etc rout right in Path, I will do the inbound routs search to find the exact match of  `/path1/path2/etc` inbound in Path. But then I suggest to do the req url change in `get_proxy_route_by_inbound` function and remove matching from Url, so we wont need to search the matching again to remove it in `modify_request_uri` function.

Reply:
> I still can do some changes on kdf side and move proxy inbound route to header

That brings the old discussion where we talk about "putting inbound route to the request". For ref, I suggest to look how other layer7 proxy servers work (e.g., nginx, traefik, etc).


> But if you prefer to have /path1/path2/etc rout right in Path, I will do the inbound routs search to find the exact match of /path1/path2/etc inbound in Path. 

In practice, we can not have every possible route registered in the configuration, unfortunately.

Reply:
> In practice, we can not have every possible route registered in the configuration, unfortunately.

Didnt get it, when we found list of matched routs, I suggest to go trough each path element until we exclude all matched elements.
what if there are two routs in config: `/nft`, `/nft/service1/`, but request matches only the first variant
`https://komodoproxy.com/nft/api/v2.2/0x1f9090aaE28b8a3dCeaDf281B0F12828e676c326/nft/transfers?chain=eth&format=decimal&order=DESC` 
We should stop at `service1`

Reply:
Done here https://github.com/KomodoPlatform/komodo-defi-proxy/pull/22/commits/64224336f8de840ec6a3297974f0846b2859d065